File: business-web/src/types/expiry.ts
/**
 * AVENZO Business Web — Expiry & Risk Types
 */

export interface ExpirySummary {
  warehouse_id?: string;
  category_id?: string;
  total_items_tracked: number;
  safe_quantity: number;
  expiring_soon_quantity: number;
  critical_quantity: number;
  expired_quantity: number;
  non_expiry_quantity: number;
  safe_batches_count: number;
  expiring_soon_batches_count: number;
  critical_batches_count: number;
  expired_batches_count: number;
}

export interface InventoryRiskMetrics {
  warehouse_id?: string;
  total_stock_quantity: number;
  near_expiry_quantity: number;
  critical_expiry_quantity: number;
  expired_quantity: number;
  expiry_exposure_percentage: number;
  capital_exposure_at_risk: number | string; // sum(quantity * cost_price)
  potential_sales_exposure: number | string; // sum(quantity * unit_price)
}

File: backend/app/schemas/expiry.py
"""
AVENZO Backend — Expiry Intelligence and Inventory Risk Schemas
"""

from uuid import UUID
from typing import Optional
from decimal import Decimal
from pydantic import BaseModel, ConfigDict


class ExpirySummaryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    warehouse_id: Optional[UUID] = None
    category_id: Optional[UUID] = None
    total_items_tracked: int
    safe_quantity: int
    expiring_soon_quantity: int
    critical_quantity: int
    expired_quantity: int
    non_expiry_quantity: int
    safe_batches_count: int
    expiring_soon_batches_count: int
    critical_batches_count: int
    expired_batches_count: int


class InventoryRiskMetricsResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    warehouse_id: Optional[UUID] = None
    total_stock_quantity: int
    near_expiry_quantity: int  # DTE <= 30 days
    critical_expiry_quantity: int  # DTE <= 7 days
    expired_quantity: int  # DTE < 0
    expiry_exposure_percentage: float  # ((near_expiry + expired) / total) * 100
    capital_exposure_at_risk: Decimal  # sum(quantity * product.cost_price) for near-expiry + expired
    potential_sales_exposure: Decimal  # sum(quantity * product.unit_price) for near-expiry + expired
